tests/test_integration.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+from lektor.admin.webui import WebUI
+from lektor.publisher import RsyncPublisher
+
+@pytest.yield_fixture(scope='function')
+def app(request, scratch_env, pad, simple_http_server):
+    original_publish = RsyncPublisher.publish
+    def fake_publish(self, target_url, credentials=None, **extra):
+        pass
+    RsyncPublisher.publish = fake_publish
```

Comment:
Another case for [pytest-mock](https://github.com/pytest-dev/pytest-mock/):

```python
def app(mocker, scratch_env, simple_http_server):
    mocker.patch("lektor.publisher.RsyncPublisher.publish")
    return WebUI(scratch_env, output_path=simple_http_server.document_root)
```

If you do it this way, you don't even need to import the `RsyncPublisher` class -- the `mock` library handles it for you. Also, I'm not sure what you need the `request` or `pad` fixtures for, so I took them out.